render/theinternet.go

```diff
@@ -46,18 +45,26 @@ var (
 	//
 	// Since names are generally <50 bytes, this shouldn't weight in
 	// at more than a few MB of memory.
-	knownServiceCache = gcache.New(10000).ARC().Build()
+	knownServiceCache        = map[string]bool{}
```

Comment:
Why not `map[string]struct{}{}`?

Reply:
It caches both positive and negative lookups, just as the previous code did.